lib/url.c

```diff
@@ -2542,6 +2551,30 @@ static CURLcode parse_proxy(struct Curl_easy *data,
     result = CURLE_OUT_OF_MEMORY;
     goto error;
   }
+#ifdef USE_UNIX_SOCKETS
+  if(sockstype && strcmp(UNIX_SOCKET_PREFIX, host) == 0) {
+    uc = curl_url_get(uhp, CURLUPART_PATH, &path, CURLU_URLDECODE);
+    if(uc) {
+      result = CURLE_OUT_OF_MEMORY;
+      goto error;
+    }
+    /* path will be "/", if no path was was found */
+    if(!(path[0]=='/' && path[1]=='\0')) {
```

Comment:
Wouldn't it be easier to just do:
~~~c
if(strcmp("/", path)) {

Reply:
Fixed. Was trying to save a function call. It should anyway be in-lined I hope.